packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -2313,18 +2383,74 @@ Hardhat Network's forking functionality only works with blocks from at least spu
 
     return { maxFeePerGas, maxPriorityFeePerGas };
   }
-}
 
-function getCommonForTracing(networkId: number, blockNumber: number): Common {
-  try {
-    const common = new Common({ chain: networkId });
+  private _selectHardfork(
+    blockNumber: BN,
+    defaultName = this._vm._common.hardfork() as HardforkName
+  ): HardforkName {
+    function assertSupported(n: HardforkName) {
+      if (!HARDHAT_NETWORK_SUPPORTED_HARDFORKS.includes(n)) {
+        throw new InternalError(`Unsupported hardfork ${n}`);
+      }
+    }
+    if (
+      this._forkBlockNumber === undefined ||
+      blockNumber.gte(new BN(this._forkBlockNumber))
+    ) {
+      assertSupported(defaultName);
```

Comment:
I don't think this assert is necessary. That's already validated.

Reply:
This value here is config.networks.hardhat.hardfork, which is already validated to be a supported one.

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/1928/commits/e05a3bc697d76b21f7a476828bbccc5d335b69dd